### ISSUE TITLE
Spanner @Column annotation setting for NOT NULL column

### ISSUE THREAD

**Issue by ChengyuanZhao** [spring-cloud/spring-cloud-gcp#1078]
This can't have an affect on the Java POJO it annotates, but merely for the schema-generation to mark those columns as not null.

**Comment by ChengyuanZhao** [comment]
@meltsufin 

I think we should add new settings to the `@Column` annotation.

`boolean GenerateSchemaNotNull` I  think it is very important to explicitly name it this way to show that it only affects the schema generation and not any java behavior.

`long GenerateSchemaParamSize` This would only be used for STRING and BYTES

`Type.code GenerateSchemaItemType` Again this only applies to generating schema, so we must make it explicitly named like this. For collection properties this refers to the item inside the collection.

**Comment by dmitry-s** [comment]
Makes sense. Maybe GenerateSchemaCollectionItemType instead of GenerateSchemaItemType ?

**Comment by meltsufin** [comment]
In JPA, `@Column` has an attribute called `nullable`, and it is only used for schema generation. I think we should stick to the same convention.
